engine/server/engine/main.go

```diff
@@ -136,30 +136,40 @@ func runMain() error {
 		return stacktrace.Propagate(err, "An error occurred getting the Kurtosis backend for backend type '%v' and config '%+v'", serverArgs.KurtosisBackendType, backendConfig)
 	}
 
-	enclaveManager, err := getEnclaveManager(kurtosisBackend, serverArgs.KurtosisBackendType, serverArgs.ImageVersionTag, serverArgs.PoolSize, serverArgs.EnclaveEnvVars, serverArgs.MetricsUserID, serverArgs.DidUserAcceptSendingMetrics)
-	if err != nil {
-		return stacktrace.Propagate(err, "Failed to create an enclave manager for backend type '%v' and config '%+v'", serverArgs.KurtosisBackendType, backendConfig)
-	}
-
-	// osFs is a wrapper around disk
 	osFs := volume_filesystem.NewOsVolumeFilesystem()
+	realTime := logs_clock.NewRealClock()
+
 	// pulls logs per enclave/per service id
 	perFileStreamStrategy := stream_logs_strategy.NewPerFileStreamLogsStrategy()
 	perFileLogsDatabaseClient := persistent_volume.NewPersistentVolumeLogsDatabaseClient(kurtosisBackend, osFs, perFileStreamStrategy)
 
 	// pulls logs /per week/per enclave/per service
-	realTime := logs_clock.NewRealClock()
 	perWeekStreamStrategy := stream_logs_strategy.NewPerWeekStreamLogsStrategy(realTime)
 	perWeekLogsDatabaseClient := persistent_volume.NewPersistentVolumeLogsDatabaseClient(kurtosisBackend, osFs, perWeekStreamStrategy)
 
+	logFileManager := log_file_manager.NewLogFileManager(kurtosisBackend, osFs, realTime)
+	go func() {
```

Comment:
I think that will be worth it to move all this threads and logic inside the logFileManager and trigger it from here, just to encapsulate the logic in the right place, but you can do it in a next PR if this one needs to get merged now

Reply:
ooo yes this is a good idea ill do this